<div align="center">

**LAPORAN PRAKTIKUM**

**ALGORITMA & PEMOGRAMAN**

**“TIPE DATA DAN VARIABLE PYTHON”**

**Disusun Oleh :**

**Muhammad Farhan Amri**

**2611533010**

**Dosen Pengampu:**

**Wahyudi.DR.. S.T.M.T**

**Asisten Praktikum:**

**Sherly Sukmadira**

<br>

 <img src="images/logo_unand.png" width="200">

<br>

**FAKULTAS TEKNOLOGI INFORMASI**

**DEPARTEMEN INFORMATIKA**

**UNIVERSITAS ANDALAS**

**2026**

</div>

## KATA PENGANTAR

Puji syukur penulis panjatkan atas kehadirat Tuhan Yang Maha Esa yang telah memberikan rahmat dan berkat-Nya, sehingga penulis dapat menyelesaikan laporan praktikum "Tipe Data dan Variabel Python" yang telah dilaksanakan sebagai pertemuan ke-2 untuk mata kuliah Praktikum Algoritma dan Pemrograman. Praktikum ini berfokus pada pemahaman pengenalan sintaks dasar Python, penggunaan tipe data dasar (numerik, string, boolean), deklarasi variabel, konstanta, serta interaksi masukan dan keluaran (I/O) melalui terminal.  
Melalui praktikum ini, hasil yang diharapkan adalah penulis dapat memahami dan mengimplementasikan secara langsung konsep dasar Python pada real project.  
Penulis menyadari bahwa laporan praktikum ini masih memiliki banyak kekurangan. Oleh karena itu, kritik dan saran yang membangun sangat diharapkan demi perbaikan dan pengembangan di masa mendatang.

Padang, September 2026

---
# BAB I
# PENDAHULUAN

## 1.1 Latar Belakang
Python menjadi salah satu bahasa pemrograman paling diminati saat ini karena sintaksnya yang ringkas dan mudah dipahami. Bagi seorang pemula di bidang informatika, menguasai cara kerja penyimpanan data di dalam memori merupakan langkah awal yang wajib dikuasai. Variabel bertindak sebagai tempat menampung nilai, sedangkan tipe data menentukan sifat dan batasan dari nilai yang disimpan tersebut. Pemahaman yang kokoh terkait alokasi variabel dan manipulasi tipe data sangat penting agar pengembangan skrip program dan penyelesaian masalah logika dapat dilakukan secara efektif.

## 1.2 Tujuan
1. Menjelaskan serta memperagakan penggunaan tipe data dasar pada Python seperti angka (integer, float, complex), teks (string), dan logika (boolean).  
2. Menerapkan teknik pemrosesan masukan pengguna (user input) beserta konversi tipe datanya (type casting).  
3. Memahami konsep variabel tetap atau nilai kaku melalui penerapan konstanta.
## 1.3 Manfaat
1. Mahasiswa terbiasa melakukan interaksi I/O sederhana dan memastikan kebenaran alur logika data.
2. Mahasiswa siap mengembangkan dan menjalankan skrip Python secara mandiri di perangkat masing-masing.

---
# BAB II
# PEMBAHASAN

## 2.1 Dasar Teori
1. Variabel dan Dynamic Typing: Di dalam Python, variabel tidak perlu dideklarasikan tipe datanya secara eksplisit. Python otomatis mendeteksi tipe data berdasarkan nilai yang diisikan ke dalam variabel tersebut.
2. Kategori Tipe Data Primitif:

A.) Numerik: Digunakan untuk perhitungan matematika. Terdiri dari int (bilangan bulat), float (bilangan desimal), dan complex (bilangan imajiner/kompleks)
.
B.) String (str): Kumpulan karakter teks yang diapit oleh tanda petik tunggal, ganda, ataupun tiga pasang petik (multiline).

C.) Boolean (bool): Tipe data yang hanya menyimpan dua nilai logika, yaitu True (benar) atau False (salah).

D.) Input data dinamis (input()): Fungsi bawaan untuk menerima ketikan dari pengguna melalui layar console. Seluruh input yang masuk akan dianggap bertipe string secara default.

E.) Konstanta: Konsep penetapan variabel yang nilainya bersifat konstan atau ditujukan agar tidak diubah-ubah sepanjang program berjalan, yang dalam Python umumnya dipertegas menggunakan penanda Final dari modul typing

## 2.2 Kode Program

### 2.2.1 Program "numerik_2611533010.py"
Program berikut mengilustrasikan penciptaan beberapa variabel, masing-masing dengan muatan tipe numerik yang berbeda (integer, float, complex), kemudian mengidentifikasi validasi struktur bawaannya memakai built-in function `type()`.

In [ ]:
# Buat file dengan nama numerik_2611533010.py
# nama variabel ditambah 4 digit terakhir NIM contoh: a_3010
# creat a variabel with integer value.
a_3010 = 100
print("The type of variable having value", a_3010, "is", type(a_3010))

# creat a variabel with float value
b_3010 = 10.2345
print("The type of variable having value", b_3010, "is ", type(b_3010))

# creat a variabel with complex value.
c_3010 = 100 + 3j
print("The type of variable having value", c_3010, " is ", type(c_3010))

The type of variable having value 100 is <class 'int'>
The type of variable having value 10.2345 is  <class 'float'>
The type of variable having value (100+3j)  is  <class 'complex'>


**Penjelasan :**
1. `a_3010 = 100` : Berfungsi untuk mendeklarasikan variabel `a_3010` dan menginisialisasinya dengan nilai bilangan bulat (integer), yaitu `100`.
2. `b_3010 = 10.23455` : Berfungsi untuk mendeklarasikan variabel `b_3010` dengan menugaskan nilai berupa pecahan atau desimal (float).
3. `c_3010 = 100+3j` : Berfungsi untuk mendeklarasikan variabel `c_3010` dengan nilai bilangan kompleks (complex) bawaan Python yang ditandai dengan sufiks `j`.
4. `type(...)` pada fungsi `print` : Berfungsi sebagai *built-in function* dari Python yang digunakan untuk mengecek tipe data aktual dari variabel yang ada di dalam tanda kurungnya, dan mencetak nama kelas tipe datanya ke layar terminal.

### 2.2.2 Program String "string_2611533010"
Blok program identitas diri di bawah ini mengandalkan string untuk mendefinisikan informasi profil. Teks dapat ditampung dalam sebaris tanda petik maupun kutipan multiline string ketika mendefinisikan lokasi alamat secara berderet.

In [4]:
# Buat file dengan nama string_2611533010.py
#nama variabel ditambah 4 digit nim terakhir contoh: nama_3010
#ganti programnya dan data masing2
nama_3010 = "Muhammad Farhan Amri"
jenis_kelamin_3010 = 'L'
alamat_3010 = """
Lubuk Lintah
kec.kuranji
kota Padang
"""
agama_3010 = 'Islam'

print("Nama saya adalah: ", nama_3010)
print("Jenis kelamin saya adalah: ", jenis_kelamin_3010)
print("Alamat saya adalah: ", alamat_3010)
print("Agama saya adalah:", agama_3010)


Nama saya adalah:  Muhammad Farhan Amri
Jenis kelamin saya adalah:  L
Alamat saya adalah:  
Lubuk Lintah
kec.kuranji
kota Padang

Agama saya adalah: Islam


Penjelasan program string di atas:

1. `nama_3010 = "Muhammad Farhan Amri"` dan `jenis_kelamin_3010 = 'L'` : Berfungsi menginisiasi variabel tipe data teks (string). Python mengizinkan pemakaian tanda kutip dua (`" "`) maupun tanda kutip tunggal (`' '`) untuk membungkus sebaris teks.
2. `alamat_3010 = """..."""` : Berfungsi untuk menampung teks yang berisi lebih dari satu baris atau *multiline string*. Teks dibungkus menggunakan tanda kutip tiga (bisa `"""` atau `'''`) agar perpindahan baris (enter) ikut tersimpan sebagai bagian dari data alamat tersebut.
3. `print("Nama saya adalah:", nama_3010)` : Berfungsi untuk mencetak gabungan string karakter statis dengan nilai yang tersimpan di dalam suatu variabel string.

### 2.2.3 Program Boolean "boolean_2611533010"
Program berikut mengaplikasikan boolean untuk mengecek syarat parameter nilai kelulusan. Status nilai akan dicari kepastiannya (`True` atau `False`) sebelum divalidasi dan diizinkan melaju memanggil perintah cetak Cum Laude.

In [5]:
# Buat file dengan nama 2611533010.py
# Nama variabel ditambah 4 digit nim terakhir contoh: nama_3010
# Deklarasi variabel dengan tipe data boolean

is_lulus_3010 = True
is_cumlaude_3010 = True

# Menggunakan Boolean
nilai_3010 = 85
batas_lulus_3010 = 75

# Menentukan nilai Boolean dari kondisi
status_kelulusan_3010 = nilai_3010 >= batas_lulus_3010  # Hasilnya akan True

print("=== Check Kelulusan ===")
print("Nilai:", nilai_3010)
print("Apakah Lulus:", status_kelulusan_3010)
if is_lulus_3010 and is_cumlaude_3010:
    print("Selamat, Anda lulus dengan predikat Cum Laude!")

=== Check Kelulusan ===
Nilai: 85
Apakah Lulus: True
Selamat, Anda lulus dengan predikat Cum Laude!


Penjelasan program boolean di atas:

1. `is_lulus_3010 = True` dan `is_cumlaude_3010 = True` : Berfungsi mendeklarasikan variabel tipe data Boolean yang merepresentasikan kebenaran, di mana nilainya hanya dapat diatur sebagai `True` (benar) atau `False` (salah).
2. `status_kelulusan_3010 = nilai_3010 >= batas_lulus_3010` : Berfungsi melakukan evaluasi operator perbandingan relasional (lebih besar sama dengan). Karena nilai 85 lebih besar dari 75, hasil evaluasi tersebut akan mengembalikan nilai kebenaran Boolean berupa `True` lalu disalurkan ke variabel `status_kelulusan_3010`.
3. `if is_lulus_3010 and is_cumlaude_3010:` : Berfungsi sebagai blok pengkondisian menggunakan operator logika `and`. Program hanya akan mengeksekusi fungsi print berupa ucapan "Selamat!" yang ada di bawahnya apabila kondisi di kiri dan kanan sintaks `and` sama-sama bernilai `True`.

### 2.2.4 Program Input "input_2611533010"
Skrip program ini bergantung pada `input()` agar sisi pengerjaan perumusan dapat secara fleksibel disuplai masukan dari sisi end-user. Masukan awal direkam menggunakan string default untuk lalu dipaksa berubah bentuk (`type casting`) menjadi bilangan `int` agar kalkulasi dimensi berjalan lancar.

In [1]:
# Buat file dengan nama input_2611533010.py
# Buat program untuk menghitung luas persegi panjang
# Nama variabel ditambah 4 digit nim terakhir contoh: nama_3010
# Program ini menggunakan fungsi input() untuk meminta pengguna
# memasukkan nilai panjang dan lebar persegi panjang.
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer
panjang_3010 = input('Masukkan nilai panjang: ')
lebar_3010 = input('Masukkan nilai lebar: ')
luas_3010 = int(panjang_3010) * int(lebar_3010)
print("Luas:", luas_3010)

Luas: 50


![Output Input](Screenshot%202026-09-15%20225542.png)



**Penjelasan :**
1. `input('Masukkan nilai panjang: ')` : Berfungsi menampilkan teks sapaan ke layar, kemudian menahan eksekusi program sejenak untuk menunggu pengguna mengetikkan sesuatu (inputan) ke dalam *console* / *terminal*. Secara bawaan (default), semua masukan akan direkam sebagai data String.
2. `int(panjang_3010)` dan `int(lebar_3010)` : Berfungsi melakukan proses *type casting* atau pemaksaan konversi tipe data.[cite: 2] Nilai panjang dan lebar yang awalnya berupa karakter teks atau String diubah paksa menjadi bilangan bulat (Integer) agar dapat dikalikan dalam operasi matematika `*` (perkalian).
3. `luas_3010 = ...` : Menyimpan hasil akhir nilai perkalian yang bertipe Integer.

### 2.2.5 Program Konstanta "konstanta_2611533010"
Menggunakan import atribut `Final` dari modul `typing`, kode ini mempatenkan variabel global `PI` senilai 3.14. Tujuannya adalah mereduksi kesalahan perubahan nilai sembarang secara tidak sengaja ketika program dikomputasi untuk menelusuri luas lingkaran.

In [1]:
# Buat file dengan nama konstanta_2611533010.py
# Program ini menggunakan konstanta untuk menghitung luas lingkaran.
# Nama variabel ditambah 4 digit nim terakhir contoh: jari_3010

from typing import Final
PI: Final = 3.14
print("pi: %f" % (PI))
jari_3010 = float(input("Masukkan nilai jari-jari: "))
luas_3010 = PI * jari_3010 * jari_3010
print("Luas lingkaran dengan jari-jari %.2f adalah %.2f" % (jari_3010, luas_3010))

pi: 3.140000
Luas lingkaran dengan jari-jari 7.00 adalah 153.86


![Output Konstanta](output_konstanta.png)

**Penjelasan :**
1. `from typing import Final` : Berfungsi untuk memanggil library atau modul bawaan `typing` dari Python, guna menggunakan kemampuan pelabelan (anotasi) jenis variabel tertentu dalam hal ini sebagai tipe `Final`.
2. `PI: Final = 3.14` : Mendeklarasikan bahwa variabel `PI` bersifat final atau konstan. Ini menunjukkan pada struktur kode bahwa `PI` merupakan tetapan nilai matematis yang tidak boleh diubah oleh baris kode mana pun secara tidak sengaja di masa mendatang.
3. `float(input(...))` : Meminta input dari pengguna, namun berkat pembungkusan fungsi tipe data `float(...)`, teks angka yang dimasukkan pengguna secara sekejap langsung dikonversi ke tipe bilangan desimal (float).
4. `%f` dan `%.2f` : Berfungsi sebagai teknik pemformatan string kuno gaya bahasa C (*String Formatting*). Simbol `%f` mencetak bilangan float secara utuh, sedangkan `%.2f` digunakan untuk membatasi tampilan hasil hitung luas dan input jari-jari agar hanya menyisakan dua digit angka spesifik saja setelah tanda koma.


Tugas Praktikum

Soal: "Sistem Biodata dan Validasi Kelulusan Praktikan"
Rancang sebuah program konsol sederhana yang mencatat data identitas praktikan baru 
dan melakukan verifikasi status kelulusan awal praktikan secara otomatis dengan syarat:

1. Nama berkas Tugas2_2611533004.py dan seluruh variabel diakhiri dengan 4 digit terakhir NIM (...3004).
2. Terapkan konstanta batas kelulusan menggunakan modul typing.Final.
3. Terapkan tipe data String & Char (untuk nama, jenis kelamin, dan multiline string untuk alamat).
4. Terapkan tipe data Numerik & Type Casting (int untuk umur, float untuk skor, dan complex untuk token identifikasi).
5. Evaluasi kelulusan secara dinamis menjadi tipe data Boolean dan tampilkan seluruh tipe data ke layar menggunakan fungsi bawaan type().

In [ ]:
Kode Jawaban
from typing import Final

# 1. Penerapan Konstanta
BATAS_LULUS: Final = 75.0

# Header Program
print("=== SISTEM REGISTRASI PRAKTIKAN ALPRO 2026 ===")

# 2. Input Data & Type Casting
nama_3010 = input("Masukkan Nama Mahasiswa   : ")
jk_3010 = input("Masukkan Jenis Kelamin (L/P): ")
umur_3010 = int(input("Masukkan Umur            : "))
skor_3010 = float(input("Masukkan Skor Tes Awal   : "))

# Alamat multiline
alamat_3010 = """Jl. Muhammad Yunus No 4,
Kecamatan Kuranji,
Kota Padang"""

# Token Sinyal (Complex)
token_3010 = 100 + 3j

# Evaluasi Status Kelulusan (Bool)
status_lulus_3010 = skor_3010 >= BATAS_LULUS

# 3. Output Data & Tipe Data
print("\n=== DATA PRAKTIKAN & HASIL PEMERIKSAAN ===")
print(f"Nama Mahasiswa : {nama_3010} | Tipe: {type(nama_3010)}")
print(f"Jenis Kelamin  : {jk_3010} | Tipe: {type(jk_3010)}")
print(f"Alamat Domisili:\n{alamat_3010}\n| Tipe: {type(alamat_3010)}")
print(f"Umur           : {umur_3010} tahun | Tipe: {type(umur_3010)}")
print(f"Skor Tes Awal  : {skor_3010} | Tipe: {type(skor_3010)}")
print(f"ID Token Sinyal: {token_3010} | Tipe: {type(token_3010)}")

print("\n=== STATUS KELULUSAN PRAKTIKUM ===")
print(f"Batas Minimum Nilai: {BATAS_LULUS}")
print(f"Apakah Dinyatakan Lulus?: {status_lulus_3010} | Tipe: {type(status_lulus_3010)}")










**Output :**


![Soal Tugas Praktikum](soal.png)



 Output: (Hasil eksekusi program (I/O) tercetak otomatis di atas sel ini menyesuaikan input saat di-run)

Penjelasan : Program tugas tugas2_3010.py adalah implementasi sistem biodata dan verifikasi kelulusan terpadu.

1. Program mengunci nilai ketetapan skor (BATAS_LULUS = 75.0) menggunakan dekorator konstan dari modul typing.Final.
2. Pengumpulan identitas memanfaatkan input() yang dikonversi bentuknya secara paksa (type casting) ke tipe numerik yang tepat: tipe integer int() untuk umur_3010 dan tipe float desimal float() untuk skor_tes_3010.
3. Variabel alamat_3010 memanfaatkan tanda kutip tiga (multiline string) agar struktur baris alamat tetap utuh berderet ke bawah, sementara variabel token_3010 diisi nilai bilangan kompleks 100+3j.
4. Evaluasi komparasi logika skor_tes_3010 >= BATAS_LULUS dilakukan untuk memeriksa apakah nilai pengguna melewati batas, lalu hasil kebenarannya disimpan sebagai True atau False (boolean) pada variabel status_lulus_3010.
5. Terakhir, variabel yang telah terisi dicetak ke layar dan dibersandingkan dengan fungsi bawaan type() guna membuktikan bahwa jenis tipe datanya valid sesuai dengan instruksi.
"""



---
# BAB III
# KESIMPULAN

3.1 Kesimpulan
Berdasarkan hasil praktikum mengenai Tipe Data dan Variabel pada Bahasa Pemrograman Python, dapat ditarik beberapa kesimpulan:
 1. Fleksibilitas Variabel: Python menggunakan sistem dynamic typing, di mana deklarasi tipe data untuk setiap variabel tidak perlu dituliskan secara manual. Python secara otomatis mengidentifikasi tipe data berdasarkan nilai yang diisikan.
 2. Keberagaman Tipe Data: Python mendukung berbagai tipe data bawaan, mulai dari numerik (int, float, complex), teks (string), hingga nilai logika (boolean), yang masing-masing memiliki peran spesifik dalam penyimpanan dan manipulasi data.
 3. Konversi Tipe Data (Type Casting): Fungsi input() secara bawaan selalu menghasilkan nilai bertipe string. Oleh karena itu, konversi ke tipe data numerik seperti int() atau float() diperlukan agar nilai masukan pengguna dapat diproses dalam perhitungan aritmatika.
 4, Penerapan Nilai Tetap (Konstanta): Nilai yang bersifat konstan dapat diamankan menggunakan modul typing (Final) untuk menjaga konsistensi nilai variabel sepanjang program berjalan,

---
## DAFTAR PUSTAKA

* Mambang, Marleny, F. D., & Zulfadhilah, M. (2022). Algoritma Pemrograman Menggunakan Python. Purwokerto: CV. Pena Persada.
   
 * Junaidi, S., & Devegi, M. (2023). Algoritma dan Pemrograman dengan Python. Pasuruan: PT. Sonpedia Publishing Indonesia.
 
 * Hamdani, I. M., & Bustamin, S. (2025). Dasar Pemrograman Python: Algoritma dan Struktur Data. Jakarta: Gramedia Digital.